Adding more framework to sample project repo

spestana opened the pull request:
I've added more details and framework in this sample project repo to aid project leaders in developing and documenting their ideas, and leading a team of hackweek participants.

spestana committed:
Update README.md

Initial updates to project template for GeoSMART hackweek

spestana committed:
Add tasks section

spestana committed:
Add model card placeholder

spestana committed:
Add model card placeholder

spestana committed:
Update model-card.md

spestana committed:
Update model-card.md

spestana committed:
Update file and folder descriptions

spestana committed:
Update README.md

spestana committed:
Update README.md

spestana committed:
Add table for collaborators

spestana committed:
Create project-checklist.md

spestana committed:
Draft ideas

spestana committed:
Update README.md

spestana committed:
Update README.md

spestana committed:
Update README.md

spestana committed:
Update README.md

spestana committed:
Update project-checklist.md

spestana committed:
Update project-checklist.md with code of conduct suggestion

spestana committed:
Fix typo

spestana committed:
Add link to AGU report

spestana committed:
Add details to tasks, methods

spestana committed:
Move link to model card tool

spestana committed:
Add code of conduct link

spestana committed:
Update with some more details

spestana committed:
Add file descriptions

spestana committed:
Delete project-checklist.md

spestana committed:
Remove checklist

spestana committed:
Update some of the text

spestana committed:
Add note about public and topics

spestana committed:
Update README.md

spestana committed:
Update explanatory text

spestana commented:
Requesting review of this PR and any other feedback ahead of the project kickoff meeting. After that meeting we'll be asking project leads to begin using this template.

@jomey @JessicaS11 @NCristea @aaarendt

aaarendt commented on README.md:
```diff
@@ -1,64 +1,99 @@
+<br> Helper utilities that are shared with the team should go in here.
 * `.gitignore`
-<br> Globally ignored files by `git` for the project.
+<br> This file sets the files that will be globally ignored by `git` for the project. (e.g. you may want git to ignore temporary files or large data files)
```
suggest linking to https://docs.github.com/en/get-started/getting-started-with-git/ignoring-files

aaarendt commented on README.md:
```diff
@@ -1,64 +1,99 @@
+* **`notebooks/`**
+<br> Notebooks that are considered delivered results for the project should go in here.
+* **`scripts/`**
+<br> Helper utilities that are shared with the team should go in here.
```
I wonder if people might think "helper" refers to our hackweek role as helper instead of a script?

It's difficult to say this succinctly but I'm thinking something like: "This is a place to store functions or subroutines that get called by multiple people. These will be files other than Jupyter Notebooks such as Python scripts (.py)."

jomey replied:
I added this folder way back then, thinking it would be helpful for people to share code. In all reality, I don't remember this getting used. Maybe it is time for this folder to be part of history.

spestana replied:
Thanks! I think I have seen it used when project leaders bring in their own scripts (e.g. for data access or pre-processing steps) to share with the team. I've updated the text to try and clarify.

aaarendt commented:
This is looking great! Is there an example from past events we could point to as a way to motivate completing the README? And are there any other ways we can entice people to fill this in? More often than not people do not fill in everything. I'll give this some more thought...

jomey commented on README.md:
```diff
@@ -1,64 +1,99 @@
+This is an example of how teams can structure their project repositories and format their project README.md file.
 
-## Files
+When creating a project repository from this template, we recommend choosing "Public" (so other participants can follow progress). You can also select Private initially and make Public before final presentation. We also recommend adding a "topic" to your repository details (click on the gear icon next to the "About" section on the repository page) to help others find your work (e.g. `uw-hackweek- ...
```
In the spirit of open science I would suggest that we only recommend "public".

spestana replied:
Do you think "You can also select Private initially and make Public before final presentation" is still fair, or also remove it? I have seen the instances where participants are shy about committing any unfinished or unpolished work because it is public, but that is something I hope we can help overcome during the hackweek :)

jomey replied:
As far as I remember, there has never been any hesitation at GeoHack, IS2, or SnowEx among participants to start public from the get go. 
Regards the unfinished work, that's why we create the personal collaborator folder, where any individual can hack away no matter what state their code is.

spestana replied:
Thanks! I've updated this to only mention creating a public repo.

spestana committed:
Addressing review comments

Clarifying description for the suggested scripts directory, add link to gitignore docs

spestana committed:
Update wording

jomey commented on README.md:
```diff
@@ -1,64 +1,99 @@
-<br> Globally ignored files by `git` for the project.
+<br> This file sets the files that will be globally ignored by `git` for the project. (e.g. you may want git to ignore temporary files or large data files, [read more about ignoring files here](https://docs.github.com/en/get-started/getting-started-with-git/ignoring-files))
 * `environment.yml`
 <br> `conda` environment description needed to run this project.
```
I wonder if we still need this too as we are all working on CryoHub.
Should we link to there instead?
Or have this optional?

spestana replied:
I am still trying to keep in mind a very generic template, so not all hackweeks will be using the same compute/cloud environment.

spestana committed:
Update instructions to suggest creating public repo

spestana commented:
@jomey or @aaarendt , if the changes look good so far, can we merge this PR so project leads can get started tomorrow after the kickoff meeting? thanks!
